### ISSUE TITLE
Bug: [consistent-generic-constructors] <short description of the issue>

### ISSUE THREAD

**Issue by yoursunny** [typescript-eslint/typescript-eslint#10445]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=DYUwLgBAhgXBCqBLAdmAHAQQE5agTwB5tc8AhAVwDNKQsAZRAaxAD4IBeCZEAdwRXTF8ACgCUAbgBQANwD2iACbQpQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQZIY6AsjgwAXxBagA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANqTokBMADJwMyQAugBpwNSPFgATAK4Zm7Lr06RRNKCTyZk0PADk5qAMIALdNADWZSlpno14vAENMAc3R4FHHv0hiAviD%2BQA&tokens=false

### Repro Code

```TypeScript
let a: Uint8Array<ArrayBufferLike> = new Uint8Array();
void a;
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-generic-constructors": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "lib": ["es2023"],
    "module": "es2022",
    "strictNullChecks": true,
    "target": "es2023"
  }
}
```

### Expected Result

Either don't report the error, or don't generate invalid code upon fix.

### Actual Result

Rule reports error:
> The generic type arguments should be specified as part of the constructor type arguments. 1:5 - 1:54

If I tap **Fix**, the code becomes:

```ts
let a = new Uint8Array<ArrayBufferLike>();
void a;
```

And then TypeScript reports error:

> 2554: Expected 1-3 arguments, but got 0. 1:9 - 1:42

### Additional Info

If I enable two rules consistent-generic-constructors and no-unnecessary-type-arguments, [playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=DYUwLgBAhgXBCqBLAdmAHAQQE5agTwB5tc8AhAVwDNKQsAZRAaxAD4IBeCZEAdwRXTF8ACgCUAbgBQANwD2iACbQpQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQZIY6AsgAacNjxFSyCtToNmbJh1qwmTRGRTIAhtEK09iWk6GwAtgPxKoVXVtAF8QEKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANqTokBMADJwMyQAugBpwNSPFgATAK4Zm7Lr06RRNKCTyZk0PADk5qAMIALdNADWZSlpno14vAENMAc3R4FHHv0hiAviD%2BQA&tokens=false), both rules would report errors.
The outcome depends on the order of fix.

* If consistent-generic-constructors is fixed first, the above problem occurs.
* If no-unnecessary-type-arguments is fixed first, the resulting code is well-formed.

**Comment by Josh-Cena** [comment]
This rule is not type-aware and cannot detect that the type `Uint8Array` has a different type signature from the value `Uint8Array`, and I don't think it's a good idea to introduce extra logic in any way to "smartly" solve this case. In the vast majority of cases, `let a: Uint8Array<ArrayBufferLike> = new Uint8Array();` and `let a = new Uint8Array<ArrayBufferLike>();` should be equivalent code, but here actually `Uint8Array()` is not a generic constructor (its overloads are), so the rule isn't even applicable.

What do others say? I don't think the TS lib def is wrong or awkward, so maybe we should add an option to ignore selected constructors?

**Comment by kirkwaiblinger** [comment]
Noting that this is a new case with 5.7, since, previously, one would have gotten the error `Type 'Uint8Array' is not generic.`. (see also [TS 5.7 releaes notes](https://devblogs.mxicrosoft.com/typescript/announcing-typescript-5-7/#typedarrays-are-now-generic-over-arraybufferlike))

I'm wondering if the `Uint8Array` constructor should be changed upstream to take a type parameter?

**Comment by Josh-Cena** [comment]
`Uint8Array` has overloaded. The form that takes no arguments is just `new () => Uint8Array<ArrayBuffer>`, which is correct in itself. You _could_ change it to `new <T extends ArrayBufferLike = ArrayBuffer> () => Uint8Array<T>` but that's extra complexity with no benefits for them.

**Comment by kirkwaiblinger** [comment]
Yeah so I guess the rule, without type information, is only really expected to work on non-overloaded class constructors. Which is most anything user-defined, and also probably a majority of the built-in types.

My intuition is that an ignore list makes sense, though `UInt8Array` and friends should probably be in it by default (or be hard-coded and concatenated). Maybe we hard-code the typed arrays, and put general ignore list up to "evaluating community engagement"?

**Comment by JoshuaKGoldberg** [comment]
Accepting PRs for specifically the built-in array-likes.

**Comment by RoenLie** [comment]
Same goes for Proxy.

![Image](https://github.com/user-attachments/assets/dde2b7f4-06d1-4bc6-a67c-46daf31ff8c4)
